### PR TITLE
Catch potential stoull exception

### PR DESCRIPTION
**Pull request by ShengguangXiao**
## Description

Catch potential stoull exception

## Backward Compatibility

- [x] This is not a breaking change
- [ ] This is a breaking change

### Implementation

- [x] **ready for review**

### Integration Test (Core Team)

- [ ] local machine test

- [ ] small-scale cloud test

### CODE REVIEW

File: src/libPersistence/Retriever.cpp [hunk 1]
```diff
@@ -256,8 +256,16 @@ bool Retriever::RetrieveBlockLink(bool trimIncompletedBlocks) {
         LOG_GENERAL(WARNING, "Get LatestActiveDSBlockNum failed");
         return false;
       }
-      m_mediator.m_ds->m_latestActiveDSBlockNum = std::stoull(
-          DataConversion::CharArrayToString(latestActiveDSBlockNumVec));
+
+      auto dsBlockNumStr =
+          DataConversion::CharArrayToString(latestActiveDSBlockNumVec);
+      try {
```
Comment (KaustubhShamshery) [inline src/libPersistence/Retriever.cpp]: The two exceptions thrown by `stoull` can be caught by `std::exception`. There is no need to use elipses.

Comment (ShengguangXiao) [reply src/libPersistence/Retriever.cpp]: Why is the try.. catch for this stoull? I cannot find it...

File: src/libPersistence/Retriever.cpp [hunk 2]
```diff
@@ -256,8 +256,16 @@ bool Retriever::RetrieveBlockLink(bool trimIncompletedBlocks) {
...
+        m_mediator.m_ds->m_latestActiveDSBlockNum = std::stoull(dsBlockNumStr);
+      } catch (...) {
```
Comment (KaustubhShamshery) [inline src/libPersistence/Retriever.cpp]:
```suggestion
      } catch (const exception& e) {
```

Comment (KaustubhShamshery) [inline src/libPersistence/Retriever.cpp]:
```suggestion
      } catch (const exception &e) {
      LOG_GENERAL(WARNING, "Exception when convert invalid DS block number "
                                 << dsBlockNumStr<<e.what());
```

File: src/libPersistence/Retriever.cpp (hunk 1 above)
Comment (ShengguangXiao) [reply src/libPersistence/Retriever.cpp]: Ok, i see. I fixed with rebase the commit, please review again. @KaustubhShamshery

**Pull request merged**